<launch>

   <arg name="namespace_1" default="robot_1"/>
   <arg name="model_1" default="vera_robot_1.urdf.xacro"/>

   <arg name="namespace_2" default="robot_2"/>
   <arg name="model_2" default="vera_robot_2.urdf.xacro"/>

   <arg name="namespace_3" default="robot_3"/>
   <arg name="model_3" default="vera_robot_3.urdf.xacro"/>

   
   <node 
      pkg="vera_mrs" type="pose_estimation.py" name="aruco_pose_publisher" output="screen">
   </node>
   
   
   <group ns="$(arg namespace_1)">
      <param name="robot_description" textfile="$(find vera_mrs)/urdf/vera_urdf/$(arg model_1)" />

      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
   </group>

   <group ns="$(arg namespace_2)">
      <param name="robot_description" textfile="$(find vera_mrs)/urdf/vera_urdf/$(arg model_2)" />

      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
   </group>

  <group ns="$(arg namespace_3)">
      <param name="robot_description" textfile="$(find vera_mrs)/urdf/vera_urdf/$(arg model_3)" />

      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  </group>



   <!--
   <node pkg="vera_mrs" type="omni_tf.py" name="omni_tf" output="screen">
   </node>
   -->
   
   <!-- Node for visualizing trajectory from turtlesim 
   <node name="robot_trajectory_visualization" pkg="vera_mrs" type="trajectory_visualization.py" output="screen">
   </node>
   -->
   

<!--
   <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
      <param name="port" value="/dev/ttyACM0"/>
      <param name="~baud" value="57600" />		
   </node>
    -->

   <!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(find vera_mrs)/urdf.rviz" required="true" />
   <! (required = "true") if rviz dies, entire roslaunch will be killed -->

</launch>
